packages/flutter/lib/src/material/data_table.dart

```diff
@@ -443,6 +443,11 @@ class DataTable extends StatelessWidget {
         textDirection: numeric ? TextDirection.rtl : null,
         children: <Widget>[ label, arrowPadding, arrow ],
       );
+    } else {
+      label = Row(
+        textDirection: numeric ? TextDirection.rtl : null,
+        children: <Widget>[ label ],
```

Comment:
having a row with just one child seems unnecessarily expensive... is this just to make sure that you can toggle onSort without losing state in the child? I guess that makes sense...

Probably better in that case to do this as one expression, with UI-as-code.

Reply:
I see your point. I debated whether to create a whole row for just one element. I went prioritized consistency over optimization. I would have no problem if that piece of code was changed, your point is valid